spec/requests/providers_spec.rb

```diff
@@ -1156,6 +1156,145 @@ def gen_import_request
     end
   end
 
+  describe 'change provider password' do
+    let(:ems_physical_infra) { FactoryGirl.create(:ems_physical_infra) }
+    let(:invalid_change_password_payload) do
+      { "action"           => "change_password",
+        "current_password" => "current_password",
+        "new_password"     => "new_password",
+        "confirm_password" => "confirm_password" }
+    end
+
+    let(:valid_change_password_payload) do
+      { "action"           => "change_password",
+        "current_password" => "current_password",
+        "new_password"     => "new_password",
+        "confirm_password" => "new_password" }
+    end
+
+    let(:valid_change_password_payload_for_multiple_providers) do
+      {
+        "action"    => "change_password",
+        "resources" => [
+          {
+            "href"             => api_provider_url(nil, ems_physical_infra),
+            "current_password" => "current_password",
+            "new_password"     => "new_password",
+            "confirm_password" => "new_password"
+          },
+          {
+            "href"             => api_provider_url(nil, ems_physical_infra),
+            "current_password" => "current_password",
+            "new_password"     => "new_password",
+            "confirm_password" => "new_password"
```

Comment:
why is there a confirm_password field in these examples ?  I expect this in a UI and not in an API.

Reply:
Like the API should be an interface with some client, I think that is important to have this field here too, providing a well know interface to password change, but yeah, it is a good discussion, where in the previous that we have we decide to keep this field, do you think that I must remove that?

Reply:
yes please, should not be required at the API level, if the model method requires it, you can default that to new_password if not specified.

btw, where is change_password_queue method defined ?  I can only see the one in 
app/models/mixins/authentication_mixin.rb which as the following signature:

```
def change_password_queue(userid, current_password, new_password, auth_type = :default)
```

unless I'm not looking at the right thing.

Reply:
You are looking at the right place, but this PR changes the method signature https://github.com/ManageIQ/manageiq/pull/17323

Reply:
but the signature of the change_password_queue method that we call in that PR has not changed. I don't see the confirm_password parameter there.